Userland/Libraries/LibWeb/CSS/StyleValues/BasicShapeStyleValue.h

```diff
@@ -7,12 +7,71 @@
 #pragma once
 
 #include <AK/Variant.h>
-#include <LibGfx/DeprecatedPath.h>
 #include <LibWeb/CSS/CSSStyleValue.h>
+#include <LibWeb/CSS/LengthBox.h>
 #include <LibWeb/CSS/PercentageOr.h>
+#include <LibWeb/CSS/StyleValues/PositionStyleValue.h>
 
 namespace Web::CSS {
 
+struct Inset {
+    Gfx::Path to_path(CSSPixelRect reference_box, Layout::Node const&) const;
+    String to_string() const;
+
+    bool operator==(Inset const&) const = default;
+
+    LengthBox inset_box;
+};
+
+struct Xywh {
+    Gfx::Path to_path(CSSPixelRect reference_box, Layout::Node const&) const;
+    String to_string() const;
+
+    bool operator==(Xywh const&) const = default;
+
+    LengthPercentage x;
+    LengthPercentage y;
+    LengthPercentage width;
+    LengthPercentage height;
+};
+
+struct Rect {
+    Gfx::Path to_path(CSSPixelRect reference_box, Layout::Node const&) const;
+    String to_string() const;
+
+    bool operator==(Rect const&) const = default;
+
+    LengthBox box;
+};
+
+enum class FitSide {
+    ClosestSide,
+    FarthestSide,
+};
```

Comment:
This suggestion is entirely optional, and apologies for not thinking of it before:

If you make this a generated enum (put it in `Enums.json`, and put `closest-side`/`farthest-side` in `Keywords.json`) then you get a couple of small benefits:
- You get a `to_string(FitSide)` function instead of doing that manually
- You also get a `keyword_to_fit_side()` function that can be used in `parse_shape_radius()`.

The second would look something like:
```c++
if (tokens.next_token().is(Token::Type::Ident)) {
    auto ident = tokens.next_token().ident();
    auto fit_side = keyword_from_string(ident).map([](auto keyword) { return keyword_to_fit_side(keyword); });
    if (!fit_side.has_value())
        return {};
    tokens.discard_a_token();
    transaction.commit();
    return *fit_side;
}
```
...which I guess isn't really less code, but it'll automatically handle any new fit-side keywords in the unlikely event that they get added. I'm probably overengineering this. :sweat_smile:

Reply:
Thanks for this idea, tried it out for a bit but I feel like it doesn't make the code much cleaner :pensive: